pkg/kubectl/resource/visitor.go

```diff
@@ -154,6 +154,11 @@ func (i *Info) Refresh(obj runtime.Object, ignoreError bool) error {
 	return nil
 }
 
+// String returns the general purpose string representation
+func (i *Info) String() string {
+	return fmt.Sprintf("namespace: %q, name: %q", i.Namespace, i.Name)
```

Comment:
This omits much of the useful info, like the resource and contained object

Reply:
Sorry for didn't respond promptly, I changed to cover more information, including `Resource\Version\Object`, now the hint would be like: 
```shell
# ./kubectl apply -f apply.yml 
error: error when retrieving current configuration of:
Resource: "pods", Version: "/v1, Kind=Pod"
Name: "", Namespace: "default"
Object: &{map["metadata":map["generateName":"testing-nnp-" "namespace":"default" "annotations":map["kubectl.kubernetes.io/last-applied-configuration":""]] "spec":map["containers":[map["command":["grep NoNewPriv /proc/self/status"] "image":"busybox" "name":"simple-container"]] "initContainers":[map["command":["grep NoNewPriv /proc/self/status"] "image":"busybox" "name":"init-container"]] "restartPolicy":"Never"] "apiVersion":"v1" "kind":"Pod"]}
from server for: "apply.yml": resource name may not be empty
```